enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,386 @@
+  - "@mrunalp"
+  - "@JoelSpeed"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@mrunalp"
+  - "@JoelSpeed"
+creation-date: 2023-05-17
+last-updated: 2023-06-22
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPNODE-1628
+---
+
+# Add CRD ImagePolicy to config.openshift.io/v1alpha1
+
+## Summary
+
+This enhancement propose an ImagePolicy CRD to provide a way to set sigstore image signature verification configuration on Openshift.
+
+## Motivation
+
+### User Stories
+
+As an OpenShift user, I want to verify the container images signed using sigstore
+tools, so that I can utilize the increased security of my software supply chain.
+
+### Goals
+
+- The user can create an ImagePolicy instance specifying the images/repositories to be verified and their policy. The MCO will write the configuration for signature verification. Once this is done, CRI-O can verify the images/repositories.
+
+- MCO container runtime config controller watches ImagePolicy instance in different namespaces and merges the instances for each namespace into a single [containers-policy.json](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) in a predefined `/path/to/policies/\<NAMESPACE\>.json`.
+
+- CRI-O can verify Cosign signature signed images using configuration from ImagePolicy by using the namespace from the sandbox config on the `PullImage` RPC.
+
+### Non-Goals
+
+- Allow configure policy for OCP product repo
+- Providing a tool to mirror the signatures is out of the scope of this enhancement. In order to verify the signature, the disconnected users need to mirror signatures together with the application images. 
+- Grant the application administrator the ability to weaken cluster-scoped policies, to avoid expanding the set of administrators capable of increasing cluster exposure to vulnerable images.
+- Grant the application administrator the ability to tighten cluster-scoped policies. This could be useful in the future, but we are deferring it to limit the amount of work needed for an initial implementation.
+
+## Proposal
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for adding signature verification configurations in
+cluster scope.
+1. The cluster administrator creates ImagePolicy yaml file with `namespace: openshift-config`.
+2. The cluster administrator creates ImagePolicy instance using `oc create -f imagepolicy.yaml`.
+3. The cluster administrator can get the merged cluster scope policies from `status` of ImagePolicy instances. 
```

Comment:
Can you more elaborate on how the status will look when two or more cluster-scoped image policies are merged? E.g. providing an example.

Reply:
I added the example that explains the result of `status.policyJSON` https://github.com/openshift/enhancements/blob/f717389a3149dd11443ba51d838853a98911669a/enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md#example-of-imagepolicy-crs
namespace scope CRs will display the policy applied within the namespace and cluster-scoped policies that override. It should be safe to display these public keys.